cvat-core/src/quality-report.ts

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+export interface RawQualityReportData {
+    id?: number;
+    parent_id?: number;
+    task_id?: number;
+    job_id?: number;
+    target: string;
+    created_date?: string;
+    gt_last_updated?: string;
+    summary?: {
+        frame_count: number,
+        frame_share: number,
+        conflict_count: number,
+        valid_count: number,
+        ds_count: number,
+        gt_count: number,
+        error_count: number,
+        warning_count: number,
+        conflicts_by_type: {
+            extra_annotation: number,
+            missing_annotation: number,
+            mismatching_label: number,
+            low_overlap: number,
+            mismatching_direction: number,
+            mismatching_attributes: number,
+            mismatching_groups: number,
+            covered_annotation: number,
+        }
+    };
+}
+
+export interface QualitySummary {
+    frameCount: number;
+    frameSharePercent: number;
+    conflictCount: number;
+    validCount: number;
+    dsCount: number;
+    gtCount: number;
+    accuracy: number;
+    precision: number;
+    recall: number;
+    errorCount: number;
+    warningCount: number;
+    conflictsByType: {
+        extraAnnotations: number;
+        missingAnnotations: number;
+        mismatchingLabel: number;
+        lowOverlap: number;
+        mismatchingDirection: number;
+        mismatchingAttributes: number;
+        mismatchingGroups: number;
+        coveredAnnotation: number;
+    }
+}
+
+export default class QualityReport {
+    public readonly id: number;
+    public readonly parentId: number;
+    public readonly taskId: number;
+    public readonly jobId: number;
+    public readonly target: string;
+    public readonly createdDate: string;
+    public readonly gtLastUpdated: string;
+    public readonly summary: QualitySummary;
+
+    constructor(initialData: RawQualityReportData) {
+        const data: RawQualityReportData = {
+            id: undefined,
+            parent_id: undefined,
+            task_id: undefined,
+            job_id: undefined,
+            target: '',
+            gt_last_updated: undefined,
+            summary: undefined,
+            created_date: undefined,
+        };
+
```

Comment:
Let's use class ``#privateProperties`` in new classes
Current approach is outdated